wasi/src/main/java/com/dylibso/chicory/wasi/WasiPreview1.java

```diff
@@ -1124,7 +1281,12 @@ public int sockSend(int sock, int siDataPtr, int siDataLen, int siFlags, int ret
     @WasmExport
     public int sockShutdown(int sock, int how) {
         logger.infof("sock_shutdown: [%s, %s]", sock, how);
-        throw new WASMRuntimeException("We don't yet support this WASI call: sock_shutdown");
+        Descriptor descriptor = descriptors.get(sock);
+        if (descriptor == null) {
+            return wasiResult(WasiErrno.EBADF);
+        }
+        // sockets are not supported, so this cannot be a socket
+        return wasiResult(WasiErrno.ENOTSOCK);
```

Comment:
Does this implementation improve the execution of some programs?
If not, I would keep the Exception for now as it's a cleaner for the user what is going on.

Reply:
This passes the tests and is a technically complete implementation. In practice, it shouldn't matter either way, as this won't be called as no program can obtain a valid socket. But if it is called with an invalid descriptor, the program will receive the expected error rather than crashing.

Reply:
There are questions that shouldn't be asked if you are not ready for the answer, thanks for explaining!